tests/include/fn_select.hpp

```diff
@@ -0,0 +1,78 @@
+/*
+ * GridTools
+ *
+ * Copyright (c) 2014-2021, ETH Zurich
+ * All rights reserved.
+ *
+ * Please, refer to the LICENSE file in the root directory.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+#pragma once
+
+#include <type_traits>
+
+#include <gridtools/meta.hpp>
+
+// fn backend
+#if defined(GT_FN_NAIVE)
+#ifndef GT_STENCIL_NAIVE
+#define GT_STENCIL_NAIVE
+#endif
+#ifndef GT_STORAGE_CPU_KFIRST
+#define GT_STORAGE_CPU_KFIRST
+#endif
+#ifndef GT_TIMER_DUMMY
+#define GT_TIMER_DUMMY
+#endif
+#include <gridtools/fn/backend2/naive.hpp>
+namespace {
+    using fn_backend_t = gridtools::fn::backend::naive;
+}
+#elif defined(GT_FN_GPU)
+#ifndef GT_STENCIL_GPU
+#define GT_STENCIL_GPU
+#endif
+#ifndef GT_STORAGE_GPU
+#define GT_STORAGE_GPU
+#endif
+#ifndef GT_TIMER_CUDA
+#define GT_TIMER_CUDA
+#endif
+#include <gridtools/fn/backend2/gpu.hpp>
+namespace {
+    template <int... sizes>
+    using block_sizes_t =
```

Comment:
Should we add a default to the backend?

Reply:
And should we move the `block_sizes_t` helper to gpu.hpp

Reply:
> Should we add a default to the backend?

Not sure. A single default value probably does not make much sense as it heavily depends on structured vs. unstructured and number of dimensions I guess.

> And should we move the `block_sizes_t` helper to gpu.hpp

Not sure either. It assumes that the domain tags are contiguous integral_constants starting at zero. But this can easily break. E.g. as soon as a user runs a 1D or 2D stencil on a 3D domain, might not hold anymore. So the more verbose ‘raw’ API might be more appropriate.